src/esw/cameras/cameras.py

```diff
@@ -0,0 +1,281 @@
+    AUTON = 0
+    ERD = 1
+    ES = 2
+    SCIENCE = 3
+
+
+class Pipeline:
+    """Every Pipeline object is control of streaming one
+    camera to its assigned ip"""
+    def __init__(
+            self, arguments: list[str],
+            current_mission_ip: str,
+            pipeline_number: int) -> None:
+        self.current_ip = current_mission_ip
+        self.device_number = -1
+        self.pipeline_number = pipeline_number
+        self.video_output = jetson.utils.videoOutput(
+            f"rtp://{self.current_ip}",
+            argv=arguments)
+        self.video_source = None
+
+    def capture_and_render_image(self) -> bool:
+        """Captures an image from the video device and streams it.
+        Returns bool representing if success or not."""
+        try:
+            image = self.video_source.Capture()
+            self.video_output.Render(image)
+            return True
+        except Exception:
+            # We may expect to catch an exception
+            # if the video source does not exist
+            return False
+
+    def get_device_number(self) -> int:
+        """Returns the camera device which the pipeline is assigned"""
+        return self.device_number
+
+    def is_currently_streaming(self) -> bool:
+        """Returns whether or not the pipeline is
+        assigned an active camera device"""
+        return self.device_number != -1
+
+    def update_device_number(
+            self, arguments: list[str], index: int) -> None:
+        """Takes in a requested camera device.
+        The pipeline will then be assigned this camera device
+        unless it does not exist. A -1 means that the pipeline
+        is no longer assigned a device."""
+        self.device_number = index
+        if index != -1:
+            self.video_source = self.video_source[index]
+            if self.video_source is not None:
+                self.video_output = jetson.utils.videoOutput(
+                    f"rtp://{self.current_ip}",
+                    argv=arguments)
+            else:
+                print(f"Unable to play camera {index} on \
+                    {self.current_ip}.")
+                self.device_number = -1
+        else:
+            self.video_source = None
+
+    def update_video_output(
+            self, arguments: list[str],
+            current_mission_ip: str) -> None:
+        """Updates the video output to ensure that pipeline
+        is streaming to the proper IP"""
+        self.current_ip = current_mission_ip
+        self.video_output = jetson.utils.videoOutput(
+                f"rtp://{self.current_ip}",
+                argv=arguments)
+
+
+class PipelineManager:
+    def __init__(self) -> None:
+        self.resolution_arguments_map: dict[str, list[str]] = {}
+        for input_map in rospy.get_param("cameras/input").items():
+            quality = input_map['resolution']
+            self.resolution_arguments_map[quality] = \

```

Comment:
I set the line limit to 119 so for stuff like this I think it makes more sense to expand.

Reply:
kk